kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -63,7 +62,7 @@ default String getKind() {
    * @return the computed {@code apiVersion} or {@code null} if neither {@link Group} or {@link Version} annotations are present
    * @throws IllegalArgumentException if only one of {@link Group} or {@link Version} is provided
    */
-  static String getApiVersion(Class<? extends HasMetadata> clazz) {
+  static String getApiVersion(Class<?> clazz) {
```

Comment:
Why? In which (valid) cases is this method going to be used with a non-HasMetadata class?

Reply:
The use case would be similar to the case above with `CustomResource.getPlural`: when we get a class that's not accurately typed but we still want to retrieve possibly existing information. Now, the question is: should the classes always be properly typed or not?

Reply:
Well, at the very least if we are dealing with some Kubernetes resource, they should implement the HasMetadata interface, most of the client expectations depend on that. (so not the same case as `Class<? extends CustomResource>`)